### PR TITLE
add acl test and advance version to 0.2.21

### PR DESCRIPTION
**Pull request by huanwwwq**

### CODE REVIEW

File: pike/smb2.py [hunk 1]
```diff
@@ -1430,6 +1448,193 @@ def _get_offset(self,owner_sid=None,group_sid=None,sacl=None,dacl=None):
...
+        self.sub_authority = []
+
+    def clone(self):
+        """
+        return a copy of this NT_SID with calculated fields reset
+        """
+        sid = NT_SID()
+        sid.revision = self.revision
+        sid.identifier_authority = self.identifier_authority
+        sid.sub_authority = self.sub_authority
```
Comment (isi-mfurer) [inline pike/smb2.py]: just noticed that this needs to be a copy

```
sid.sub_authority = list(self.sub_authority)
```

or 

```
sid.sub_authority = self.sub_authority[:]
```

**Comment by isi-mfurer** [comment]
i did my checks and i'm ready to merge, once the sub_authority issue is resolved.

File: pike/smb2.py (hunk 1 above)
Comment (huanwwwq) [reply pike/smb2.py]: fixed

**Comment by isi-mfurer** [comment]
Tested internally with helix was was successful. Taking merge

**Pull request merged**